pkg/nfd-master/nfd-master.go

```diff
@@ -1310,11 +1319,15 @@ func (m *nfdMaster) nfdAPIUpdateHandlerWithLeaderElection() {
 		RenewDeadline: m.config.LeaderElection.RenewDeadline.Duration,
 		Callbacks: leaderelection.LeaderCallbacks{
 			OnStartedLeading: func(_ context.Context) {
-				m.nfdAPIUpdateHandler()
+				m.isLeader = true
 			},
 			OnStoppedLeading: func() {
 				// We lost the lock.
 				klog.InfoS("leaderelection lock was lost")
+				// Exit, makes sure that in-flight requests/re-tries will be
```

Comment:
Shouldn't we `isLeader = false` when stopping being the leader?

Reply:
I thought about adding it, but put it in the TODO comment instead. We do stop() and thus exit the process so isLeader=false would be kinda nop. I can add it if you think it's better that way.

Reply:
I think is a good first approach, I am good to lgtm